src/nvim/api/buffer.c

```diff
@@ -108,7 +108,7 @@ ArrayOf(String) buffer_get_line_slice(Buffer buffer,
   Array rv = ARRAY_DICT_INIT;
   buf_T *buf = find_buffer_by_handle(buffer, err);
 
-  if (!buf) {
+  if (!buf || !inbounds(buf, start)) {
```

Comment:
With this change, how does one gets the last N lines of a buffer without making two msgpack-rpc requests?